.github/workflows/platforms.yml

```
@@ -0,0 +1,118 @@
+             uses: actions/checkout@v3
+           - name: "Setup latest Alpine Linux"
+             uses: jirutka/setup-alpine@v1
+             with:
+               arch: x86
+               branch: edge
+               packages: >
+                 llvm-dev
+                 clang-dev
+                 clang-static
+                 llvm17-static
+                 llvm17-gtest
+                 cmake
+                 llvm
+                 clang
+                 make
+                 git
+           - name: "Setup"
+             run: |
+               clang --version
+               make --version
+             shell: alpine.sh {0}
+
+           - name: "Compile library"
+             run: |
+               mkdir build && cd build
+               cmake -DLLVM_DIR=/usr/lib/ -DCMAKE_BUILD_TYPE=DEBUG -DLLVM_EXTERNAL_LIT="$(which lit)" ../../clad
+               make -j 8 check-clad
+             shell: alpine.sh {0}
+
+
+   build_2:
+       runs-on: ubuntu-latest
+       name: armv7
+       steps:
+           - name: Checkout branch
+             uses: actions/checkout@v3
+           - name: "Setup latest Alpine Linux"
+             uses: jirutka/setup-alpine@v1
+             with:
+               arch: armv7
+               branch: edge
+               packages: >
+                 llvm-dev
+                 clang-dev
+                 clang-static
+                 llvm17-static
+                 llvm17-gtest
+                 cmake
+                 llvm
+                 clang
+                 make
+                 git
+           - name: "Setup"
+             run: |
+               clang --version
+               make --version
+             shell: alpine.sh {0}
+
+           - name: "Compile library"
+             run: |
+               mkdir build && cd build
+               cmake -DLLVM_DIR=/usr/lib/ -DCMAKE_BUILD_TYPE=DEBUG -DLLVM_EXTERNAL_LIT="$(which lit)" ../../clad
+               make -j 8 check-clad
+             shell: alpine.sh {0}
+
+
+   build_3:
+       runs-on: ubuntu-latest
+       name: aarch64
+       steps:
+           - name: Checkout branch
+             uses: actions/checkout@v3
+           - name: "Setup latest Alpine Linux"
+             uses: jirutka/setup-alpine@v1
+             with:
+               arch: aarch64
+               branch: edge
+               packages: >
+                 llvm-dev
+                 clang-dev
+                 clang-static
+                 llvm17-static
+                 llvm17-gtest
+                 cmake
+                 llvm
+                 clang
+                 make
+                 git
+           - name: "Setup"
+             run: |
+               clang --version
+               make --version
+             shell: alpine.sh {0}
+
+           - name: "Compile library"
+             run: |
+               mkdir build && cd build
+               cmake -DLLVM_DIR=/usr/lib/ -DCMAKE_BUILD_TYPE=DEBUG -DLLVM_EXTERNAL_LIT="$(which lit)" ../../clad
```

```suggestion
               cmake -DLLVM_EXTERNAL_LIT="$(which lit)" ../../clad
```

Can we extract the step "Compile library" into one place as it seems to repeat across builds. The same for the other common steps. Maybe [this](https://stackoverflow.com/questions/59757355/reuse-portion-of-github-action-across-jobs) could help?

---

Sorry for the delayed response. Yes, I will look into this and patch it asap.

---

Can you take a look how the clad main workflow is organized, we want something similar for this PR.

---

Yes, will do this.
Thanks!

---

Hi, I have updated with respect to clad convention adhering to matrix strategy.
Thanks!